#!/usr/bin/env python
PACKAGE = "robotics_project_one"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

integrationMethod_enum = gen.enum ([gen.const("Euler", int_t,0 , "Euler method") ,
                                     gen.const("RK",int_t, 1, "Runge-kutta method")],
                                      "type of method of integration")

gen.add("integrationMethod", int_t, 0, "Odometry method to use", 0, 0, 1, edit_method = integrationMethod_enum)

gen.add("r", double_t , 1 , "robot wheel radius", 0.07)
gen.add("N", double_t , 2, "robot CPR", 42)


exit(gen.generate(PACKAGE, "compute_odometry", "odometryIntegration"))   



   